### PR TITLE
fix(windows): more reliable extracting

### PR DESCRIPTION
**Pull request by paperclover**
Windows cannot move a directory while we have a handle to it open

The zig standard library also aquires file locks.

We also close a file descriptor to only re-open it for renmaing (posix doesn't have this type of "rename open file handle", but windows does, so use it!)

Some other misc changes i did while messing with other parts of codebase.

### CODE REVIEW

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
does not compile on posix

File: src/install/extract_tarball.zig [hunk 1]
```diff
@@ -432,3 +441,92 @@ fn extract(this: *const ExtractTarball, tgz_bytes: []const u8) !Install.ExtractD
...
+    switch (rc) {
+        .SUCCESS => return result,
+        .OBJECT_NAME_INVALID => return error.BadPathName,
+        .OBJECT_NAME_NOT_FOUND => return error.FileNotFound,
+        .OBJECT_PATH_NOT_FOUND => return error.FileNotFound,
+        .NOT_A_DIRECTORY => return error.NotDir,
+        // This can happen if the directory has 'List folder contents' permission set to 'Deny'
+        // and the directory is trying to be opened for iteration.
+        .ACCESS_DENIED => return error.AccessDenied,
+        .INVALID_PARAMETER => unreachable,
```
Comment (Jarred-Sumner) [inline src/install/extract_tarball.zig]:
```suggestion
        .INVALID_PARAMETER => return error.EINVAL,
```

File: src/libarchive/libarchive.zig [hunk 2]
```diff
@@ -564,7 +562,7 @@ pub const Archive = struct {
                         Kind.sym_link => {
                             const link_target = lib.archive_entry_symlink(entry).?;
                             if (comptime Environment.isWindows) {
-                                @panic("TODO on Windows");
+                                @panic("TODO on Windows: Extracting archives containing symbolic links.");
```
Comment (Jarred-Sumner) [inline src/libarchive/libarchive.zig:565]: We should just support this. npm packages do it all the time.

Comment (paperclover) [reply src/libarchive/libarchive.zig:565]: sure. Does NPM/Yarn/PNPM have a fallback for if symbolic links are not allowed?

if so, we need to implement that in addition to trying the symlink.

Comment (Jarred-Sumner) [reply src/libarchive/libarchive.zig:565]: probably reparse points

File: src/cli/create_command.zig [hunk 3]
```diff
@@ -2255,18 +2255,30 @@ const GitHandler = struct {
...
+        // Benchmark #1: bun create react3 app --force --no-install
+        //   Time (mean ± σ):     306.7 ms ±   6.1 ms    [User: 31.7 ms, System: 269.8 ms]
+        //   Range (min … max):   299.5 ms … 318.8 ms    10 runs
 
         if (which(&bun_path_buf, PATH, destination, "git")) |git| {
             const git_commands = .{
-                &[_]string{ bun.asByteSlice(git), "init", "--quiet" },
-                &[_]string{ bun.asByteSlice(git), "add", destination, "--ignore-errors" },
-                &[_]string{ bun.asByteSlice(git), "commit", "-am", "Initial commit (via bun create)", "--quiet" },
+                &[_]string{ git, "init", "--quiet" },
```
Comment (Jarred-Sumner) [inline src/cli/create_command.zig:2279]: this is a usecase for the shell!

File: src/install/extract_tarball.zig [hunk 4]
```diff
@@ -195,7 +209,8 @@ fn extract(this: *const ExtractTarball, tgz_bytes: []const u8) !Install.ExtractD
             return error.InstallFailed;
         };
 
-        defer extract_destination.close();
+        errdefer if (Environment.isWindows) extract_destination.close();
```
Comment (Jarred-Sumner) [inline src/install/extract_tarball.zig:212]: double close on `errdefer`.

Comment (Jarred-Sumner) [reply src/install/extract_tarball.zig:212]: nvm actually

File: src/install/install.zig [hunk 5]
```diff
@@ -692,7 +692,9 @@ const Task = struct {
                 ) catch |err| {
                     if (comptime Environment.isDebug) {
                         if (@errorReturnTrace()) |trace| {
-                            std.debug.dumpStackTrace(trace.*);
+                            _ = trace; // autofix
```
Comment (Jarred-Sumner) [inline src/install/install.zig:695]: ?

**Pull request merged**